pkg/virtctl/ssh/ssh.go

```diff
@@ -0,0 +1,173 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package ssh
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/kubevirt/pkg/virtctl/templates"
+)
+
+const (
+	portFlag, portFlagShort                         = "port", "p"
+	wrapLocalSSHFlag                                = "local-ssh"
+	usernameFlag, usernameFlagShort                 = "username", "l"
+	identityFilePathFlag, identityFilePathFlagShort = "identity-file", "i"
+	knownHostsFilePathFlag                          = "known-hosts"
+)
+
+var (
+	wrapLocalSSH              bool = false
+	sshPort                   int
+	sshUsername               string
+	identityFilePath          string
+	knownHostsFilePath        string
+	knownHostsFilePathDefault string
+)
+
+func NewCommand(clientConfig clientcmd.ClientConfig) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "ssh (VM|VMI)",
+		Short:   "Open a SSH connection to a virtual machine instance.",
+		Example: usage(),
+		Args:    templates.ExactArgs("ssh", 1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c := SSH{clientConfig: clientConfig}
+			return c.Run(cmd, args)
+		},
+	}
+
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		glog.Warningf("failed to determine user home directory: %v", err)
+	}
+	if len(homeDir) > 0 {
+		knownHostsFilePathDefault = filepath.Join(homeDir, ".ssh", "known_hosts")
+	}
+
+	cmd.Flags().StringVar(&sshUsername, usernameFlag, "",
+		fmt.Sprintf("--%s=jdoe: Set this to the user you want to open the SSH connection as; If unassigned, this will be empty and the SSH default will apply", usernameFlag))
+	cmd.Flags().StringVarP(&identityFilePath, identityFilePathFlag, identityFilePathFlagShort, "",
```

Comment:
That means right now we wouldn't even try the SSH agent.
I got to try if I can add both and it will favor the agent even if a key file is specified.
Right now specifying a key is a way to skip the agent.

Reply:
Done, all auth methods now work in order.